cmake/tools.cmake

```diff
@@ -53,11 +53,14 @@ list (GET COMPILER_VERSION_LIST 0 COMPILER_VERSION_MAJOR)
 # Example values: `lld-10`, `gold`.
 option (LINKER_NAME "Linker name or full path")
 
-if (NOT LINKER_NAME)
-    if (COMPILER_GCC)
+# s390x doesnt support lld
+if (COMPILER_GCC)
```

Comment:
Perhaps my code snippet in https://github.com/ClickHouse/ClickHouse/pull/40988#discussion_r971253740 was a bit misleading but we should not drop l. 56 (left side), i.e. `if (NOT LINKER_NAME)` here. Otherwise, we'll call `find_program` even if a linker was specified explicitly.

Future readers will also wonder why l. 56 speaks about "lld" but the "if"-s in l. 58 and l. 63 refer to "lld" and "gold". Please either update the comment (if s390x also doesn't support gold) or limit the scope of the "if"-s to "lld" as proposed in https://github.com/ClickHouse/ClickHouse/pull/40988#discussion_r971253740.